ndb/src/google/cloud/ndb/_eventloop.py

```diff
@@ -259,6 +263,64 @@ def run(self):
                 break
 
 
+class _LocalContexts(threading.local):
+    """Maintain a thread local stack of event loops."""
+
+    def __init__(self):
+        self.stack = []
+
+    def push(self, loop):
+        self.stack.append(loop)
+
+    def pop(self):
+        return self.stack.pop(-1)
+
+    def current(self):
+        if self.stack:
+            return self.stack[-1]
+
+
+contexts = _LocalContexts()
+
+
+@contextlib.contextmanager
+def async_context():
+    """Establish an asynchronous context for a set of asynchronous API calls.
+
+    This function provides a context manager which establishes the event loop
+    that will be used for any asynchronous NDB calls that occur in the context.
```

Comment:
I just realized I have no clear picture in my head of how we should refer to "NDB" / `ndb` / "Ndb" / etc.

Do you have a preference?

Reply:
Acronyms are generally capitalized. Google calls it [NDB](https://cloud.google.com/appengine/docs/standard/python/ndb/). NDB.

Reply:
:+1: